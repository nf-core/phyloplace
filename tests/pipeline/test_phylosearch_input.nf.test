nextflow_pipeline {

    name "Test Workflow main.nf"
    script "main.nf"
    tag "test_phylosearch_input"
    tag "pipeline"

    test("phylosearch input csv") {
        config "./nextflow.config"
        when {
            params {
                outdir            = "$outputDir"
                phylosearch_input = 'https://raw.githubusercontent.com/nf-core/test-datasets/phyloplace/testdata/phylosearch_input.csv'
                search_fasta      = 'https://raw.githubusercontent.com/nf-core/test-datasets/phyloplace/testdata/domain_16s.fna'
                max_cpus          = 2
                max_memory        = '6.GB'
                max_time          = '6.h'
            }
        }

        then {
            assertAll(
                { assert workflow.success },
                { assert new File("$outputDir/hmmer/rank.hmmrank.tsv.gz").exists() },
                { assert snapshot("$outputDir/hmmer/rank.hmmrank.tsv.gz").match('rank.hmmrank.tsv.gz') },
                { assert new File("$outputDir/epang/arc.epa_result.jplace.gz").exists() },
                { assert new File("$outputDir/epang/arc.query.fasta.gz").exists() },
                { assert new File("$outputDir/epang/arc.reference.fasta.gz").exists() },
                { assert new File("$outputDir/gappa/arc.graft.arc.epa_result.newick").exists() },
                { assert snapshot("$outputDir/gappa/arc.graft.arc.epa_result.newick").match('arc.graft.newick') },
                { assert new File("$outputDir/gappa/arc.taxonomy.per_query.tsv").exists() },
                { assert new File("$outputDir/clustalo/arc.aln").exists() },
                { assert snapshot("$outputDir/clustalo/arc.aln").match('arc.aln') },
                { assert new File("$outputDir/epang/bac16s.epa_result.jplace.gz").exists() },
                { assert new File("$outputDir/epang/bac16s.query.fasta.gz").exists() },
                { assert new File("$outputDir/epang/bac16s.reference.fasta.gz").exists() },
                { assert new File("$outputDir/gappa/bac16s.graft.bac16s.epa_result.newick").exists() },
                { assert snapshot("$outputDir/gappa/bac16s.graft.bac16s.epa_result.newick").match('bac16s.graft.newick') },
                { assert new File("$outputDir/gappa/bac16s.taxonomy.per_query.tsv").exists() },
                { assert new File("$outputDir/clustalo/bac16s.aln").exists() },
                { assert snapshot("$outputDir/clustalo/bac16s.aln").match('bac16s.aln') },
                { assert new File("$outputDir/pipeline_info/nf_core_phyloplace_software_mqc_versions.yml").exists() },
                { assert new File("$outputDir/multiqc/multiqc_report.html").exists() },
            )
        }
    }
}
