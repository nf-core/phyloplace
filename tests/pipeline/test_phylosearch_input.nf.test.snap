{
    "arc.graft.newick": {
        "content": [
            "arc.graft.arc.epa_result.newick:md5,856b750cce92b19c0a469862717915a2"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-11T19:10:28.064875241"
    },
    "bac16s.graft.newick": {
        "content": [
            "bac16s.graft.bac16s.epa_result.newick:md5,d2a349ff8796149635c164ca4f9fa376"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-11T19:12:13.343567827"
    },
    "arc.aln": {
        "content": [
            "arc.aln:md5,48e87b305ac10ecc0036ba7ab1c84c86"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-11T19:05:04.206028125"
    },
    "rank.hmmrank.tsv.gz": {
        "content": [
            "rank.hmmrank.tsv.gz:md5,2befde6dc561a046938737b94c0a3a6e"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-11T19:08:50.587132125"
    },
    "bac16s.aln": {
        "content": [
            "bac16s.aln:md5,188fcf5c9096295c04d88619f1f3c7da"
        ],
        "meta": {
            "nf-test": "0.9.0",
            "nextflow": "24.10.2"
        },
        "timestamp": "2024-12-11T19:08:50.611653261"
    }
}